src/signing.js

```diff
@@ -101,66 +95,93 @@ class SignatureFactory {
   }
 
   canonicalString() {
+    /**
+     * HTTPRequestMethod
+     * CanonicalURI
+     * CanonicalQueryString
+     * CanonicalHeaders
+     * SignedHeaders
+     * HexEncode(Hash(RequestPayload))
+     */
     return [
-      this.method,                // HTTPRequestMethod
-      this.url,                   // CanonicalURI
-      this.queryParameters,       // CanonicalQueryString
-      this.getCanonicalHeaders(), // CanonicalHeaders
-      this.getSignedHeaders(),    // SignedHeaders
-      this.hash('', 'hex')        // HexEncode(Hash(RequestPayload))
+      this.method,
+      this.url,
+      this.queryParameters,
+      this.getCanonicalHeaders(),
+      this.getSignedHeaders(),
+      SignatureFactory.hash('', 'hex')
     ].join('\n');
   };
 
-  hmac(key, string, encoding) {
-    return crypto.createHmac('sha256', key).update(string, 'utf8').digest(encoding);
-  }
-
   stringToSign() {
+    /**
+     * Algorithm
+     * RequestDate
+     * CredentialScope
+     * HashedCanonicalRequest
+     */
     return [
-      'USBL1-HMAC-SHA256',                          // Algorithm
-      this.dates.longdate,                          // RequestDate
-      this.dates.shortdate + '/' + 'usbl1_request', // CredentialScope
-      this.hash(this.canonicalString(), 'hex')      // HashedCanonicalRequest
+      'USBL1-HMAC-SHA256',
+      this.dates.longdate,
+      this.dates.shortdate + '/' + 'usbl1_request',
+      SignatureFactory.hash(this.canonicalString(), 'hex')
     ].join('\n');
   };
 
   getSignature() {
-    const kDate = this.hmac('USBL1' + this.secretKey, this.dates.shortdate);
-    const kSigning = this.hmac(kDate, 'usbl1_request');
+    const kDate = SignatureFactory.hmac('USBL1' + this.secretKey, this.dates.shortdate);
+    const kSigning = SignatureFactory.hmac(kDate, 'usbl1_request');
 
-    return this.hmac(kSigning, this.stringToSign(), 'hex');
+    return SignatureFactory.hmac(kSigning, this.stringToSign(), 'hex');
   }
 
   authHeader() {
-    this.dates = this.getDateTime();
+    this.dates = SignatureFactory.getDateTime();
     this.headers['x-usbl-date'] = this.dates.longdate;
 
     return [
-      'USBL1-HMAC-SHA256 Credential=' + this.accessKey + '/' + this.dates.shortdate + '/' + 'usbl1_request',
-      'SignedHeaders=' + this.getSignedHeaders(),
-      'Signature=' + this.getSignature()
+      `USBL1-HMAC-SHA256 Credential=${this.accessKey}/${this.dates.shortdate}/usbl1_request`,
+      `SignedHeaders=${this.getSignedHeaders()}`,
+      `Signature=${this.getSignature()}`
     ].join(', ');
   };
 
-  getDateTime() {
+  sign() {
+    this.headers['Authorization'] = this.authHeader();
+
+    return {
+      headers: this.headers,
+      url: (this.queryParameters) ? `${this.url}?${this.queryParameters}` : this.url
+    };
+  }
+
+  static getDateTime() {
```

Comment:
I like it when the static methods defined before the constructor, instead of on the bottom of the class. What do you think?